src/main/java/org/jenkinsci/plugins/p4/PerforceScm.java

```diff
@@ -220,8 +224,22 @@ public String getKey() {
 			logger.fine("No credential for perforce");
 			return null;
 		}
+
+		// Retrieve item from request
+		StaplerRequest req = Stapler.getCurrentRequest();
+		Job job = req == null ? null : req.findAncestorObject(Job.class);
```

Comment:
> And why are you doing this calculation now? Do you actually need to contact the server in order to determine whether to use SwarmBrowser (whatever that is) or something else?

Apparently, the Swarm URL is a server property `P4.Swarm.URL` so you need to connect to the server in order to retrieve it. Also the server URI is stored in P4 Credentials.

The current implementation for `guessBrowser` is to lookup the credentials specified for this SCM at Jenkins root. Given the context that Jenkins provides. it seems that P4 can simply not properly guess the browser implementation and simply return `null`. Is that correct ?